Crud.php

```diff
@@ -0,0 +1,271 @@
+        const ENTITY_NAME = "Number"; // NAme of entity this controller handle
+        const ENTITY_PATH = __DIR__."/src/".self::ENTITY_NAME; // PAth to entity file
+        const ENTITY_ALIAS = "src\Number"; // Alias for entity
+
+        /**
+         * Method for creating of class
+         * 
+         * Takes entyty manager instance to work with doctrine
+         * entity manager instance is generated inside bootstrap file
+         * 
+         * @param entityManager
+         */
+        public function __construct($entityManager) {
+            $this->entityManager = $entityManager;
+        }
+
+        /**
+         * Adds new number
+         * 
+         * @param string slug
+         * @param string title
+         * @param int number
+         * @param string text
+         * @param string transcription
+         * 
+         * @return bool
+         */
+        public function createNumber(string $slug, string $title, int $number, string $text, string $transcription):bool {
+            $numberEntity = new Number();
+            $numberEntity->setSlug($slug);
+            $numberEntity->setTitle($title);
+            $numberEntity->setNumber($number);
+            $numberEntity->setText($text);
+            $numberEntity->setTranscription($transcription);
+            $numberEntity->setDate(new DateTime());
+            $this->entityManager->persist($numberEntity);
+            $this->entityManager->flush();
+
+            return true;
+        }
+
+        /**
+         * Read number data
+         * 
+         * @param int id
+         * 
+         * @return object
+         */
+        public function readNumber(int $id):object {
+            $number = $this->entityManager->find('Number', $id);
+
+            return $number;
+        }
+
+        /**
+         * Search numbers
+         * 
+         * @param slug
+         * @param title
+         * @param text
+         * @param transcription
+         * @param date
+         */
+        public function findNumbers($slug = null, $title = null, $text = null, $transcription = null, $date = null) {}
+
+        /**
+         * Read all numbers
+         * 
+         * @return object
+         */
+        public function readAllNumbers():object {
+            $numberRepository = $this->entityManager->getRepository('Number');
+            $numbers = $numberRepository->findAll();
+
+            return $numbers;
+        }
+
+        /**
+         * Reads numbers paginated
+         * 
+         * @param int page
+         * 
+         * @return array
+         */
+        public function readNumbersPaginate(int $page = 1):array {               
+            // get the user repository
+            $numbers =  $this->entityManager->getRepository(Number::class);
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n')
+                ->getQuery();
+
+            //set page size
+            $pageSize = self::PER_PAGE;
```

Comment:
I believe, $pageSize variable here is redundund.  You could use self:PER_PAGE constant here.

Reply:
Yes I created that constant for this purpose. Will change.